drizzlepac/alignimages.py

```diff
@@ -1134,188 +1215,208 @@ def update_image_wcs_info(tweakwcs_output):
-                ra_rms = np.std(dra.to(u.mas))
-                dec_rms = np.std(ddec.to(u.mas))
-                fit_rms = np.std(Angle(img_coords.separation(ref_coords), unit=u.mas)).value
-                group_dict[group_id]['FIT_RMS'] = fit_rms
-                group_dict[group_id]['RMS_RA'] = ra_rms
-                group_dict[group_id]['RMS_DEC'] = dec_rms
-
-                obs_rms.append(fit_rms)
-    # Compute RMS for entire ASN/observation set
-    total_rms = np.mean(obs_rms)
-    # total_rms = np.sqrt(np.sum(np.array(obs_rms)**2))
+    grouped_tpwcs = OrderedDict([(None, [])])
+    for tpwcs in tpwcs_lst:
+        group_id = tpwcs.meta['group_id']
+        if group_id in grouped_tpwcs:
+            grouped_tpwcs[group_id].append(tpwcs)
+        else:
+            grouped_tpwcs[group_id] = [tpwcs]
+    non_grouped = grouped_tpwcs.pop(None)
+
+    num_fits = 0
+    total_rms = 0.0
+
+    # iterate over "groups":
+    for tp_items in grouped_tpwcs.values():
+        tpwcs = tp_items[0]
+        tinfo = tpwcs.meta['fit_info']
+        if tinfo['status'] == 'SUCCESS':
+            log.info("fit_info: {}".format(tinfo))
+            fit_rms, ra_rms, dec_rms = _compute_fit_sky_rms(
+                tinfo, reference_catalog
+            )
+            num_fits += 1
+            total_rms += fit_rms
 
-    # Now, append computed results to tweakwcs_output
-    for item in tweakwcs_output:
-        group_id = item.meta['group_id']
-        if group_id in group_dict:
-            fit_rms = group_dict[group_id]['FIT_RMS']
-            ra_rms = group_dict[group_id]['RMS_RA']
-            dec_rms = group_dict[group_id]['RMS_DEC']
         else:
-            fit_rms = None
-            ra_rms = None
-            dec_rms = None
+            # fit info is not complete when fit status is other than 'SUCCESS'
+            fit_rms, ra_rms, dec_rms = None
+
+        for tpwcs in tp_items:
+            tpwcs.meta['fit_info']['FIT_RMS'] = fit_rms
+            tpwcs.meta['fit_info']['RMS_RA'] = ra_rms
+            tpwcs.meta['fit_info']['RMS_DEC'] = dec_rms
+
+    # iterate over non-grouped/stand-alone WCS:
+    for tpwcs in non_grouped:
+        tinfo = tpwcs.meta['fit_info']
+        if tinfo['status'] == 'SUCCESS':
+            log.info("fit_info: {}".format(tinfo))
+            fit_rms, ra_rms, dec_rms = _compute_fit_sky_rms(
+                tinfo, reference_catalog
+            )
+            num_fits += 1
+            total_rms += fit_rms
 
-        item.meta['fit_info']['FIT_RMS'] = fit_rms
-        item.meta['fit_info']['TOTAL_RMS'] = total_rms
-        item.meta['fit_info']['NUM_FITS'] = len(group_ids)
-        item.meta['fit_info']['RMS_RA'] = ra_rms
-        item.meta['fit_info']['RMS_DEC'] = dec_rms
-        item.meta['fit_info']['catalog'] = reference_catalog.meta['catalog']
+        else:
+            # fit info is not complete when fit status is other than 'SUCCESS'
+            fit_rms, ra_rms, dec_rms = None
```

Comment:
Same comment as for Line 1386.

Reply:
@mdlpstsci I fixed most except logging context manager issue. I need to investigate that one.